models/instrument.py

```diff
@@ -1087,6 +1129,148 @@ def get_short_instrument_name(cls):
         """
         return 'Demo'
 
+    def find_origin_exposures( self, skip_exposures_in_database=True,
+                               minmjd=None, maxmjd=None, filters=None,
+                               containing_ra=None, containing_dec=None,
+                               minexptime=None ):
+        """Search the external repository for this instrument.
+
+        Search the external image/exposure repository for this
+        instrument for exposures that the database doesn't know about
+        already.  For example, for DECam, this searches the noirlab data
+        archive.
+
+        WARNING : do not call this without some parameters that limit
+        the search; otherwise, too many things will be returned, and the
+        query is likely to time out or get an error from the external
+        repository.
+
+        Parameters
+        ----------
+        skip_exposures_in_databse: bool
+           If True (default), will filter out any exposures that (as
+           best can be determined) are already known in the SeeChange
+           database.  If False, will include any 
+        minmjd: float
+           The earliest time of exposure to search (default: no limit)
+        maxmjd: float
+           The latest time of exposure to search (default: no limit)
+        filters: str or list of str
+           Filters to search.  The actual strings are
+           instrument-dependent, and will match what is expected on the
+           external repository.  By default, doesn't limit by filter.
+        containing_ra: float
+           Search for exposures that include this RA (degrees, J2000);
+           default, no RA constraint.
+        containing_dec: float
+           Search for exposures that include this Dec (degrees, J2000);
+           default, no Dec constraint.
+        minexptime: float
+           Search for exposures that have this minimum exposure time in
+           seconds; default, no limit.
+
+        Returns
+        -------
+        A InstrumentOriginExposures object, or None if nothing is found.
+
+        """
+        raise NotImplementedError( f"Instrument class {self.__class__.__name__} hasn't "
+                                   f"implemented find_origin_exposures." )
+
+class InstrumentOriginExposures:
+    """A class encapsulating the response from Instrument.find_origin_exposures()
+
+    Never instantiate one of these (or a subclass) directly; get it from
+    find_origin_exposures().
+
+    Must be subclassed by each instrument that defines
+    find_origin_exposures().  The internal storage of the exposures will
+    differ for each instrument, and no external assumptions should be
+    made about it other than that it's a sequence.
```

Comment:
Something that would be an instance of collections.abc.Sequence -- lists, tuples, strings.

The real assumption here is that's something that can be indexed with an integer, because some of the other methods take an integer as an argument.

Reply:
I mean to ask "is the internal storage supposed to be a sequence?" the subject in that sentence isn't clear...